### PR TITLE
✨(api) make username editable via the user API

### PR DESCRIPTION
**Pull request by sampaccoud**
## Purpose

We want to allow changing one's username. This is acceptable as long as the field remains unique.

## Proposal

Remove `username` from the list of readonly fields

### CODE REVIEW

File: src/magnify/apps/core/models.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ class User(BaseModel, AbstractBaseUser, PermissionsMixin):
     """
 
     username_validator = RegexValidator(
-        "^[a-zA-Z0-9_-]*$",
+        "^[a-z0-9_-]*$",
         message=_(
             "Username must contain only letters, numbers, underscores and hyphens."
```
Comment (kernicPanel) [inline src/magnify/apps/core/models.py:57]:
```suggestion
            "Username must contain only lower case letters, numbers, underscores and hyphens."
```

**Review by kernicPanel — COMMENTED** [review]
Just a suggestion regarding helper text, otherwise LGTM

File: src/magnify/apps/core/models.py (hunk 1 above)
Comment (sampaccoud) [reply src/magnify/apps/core/models.py:57]: Good idea :pray:

**Review by kernicPanel — COMMENTED** [review]
IMHO, a migration should downcase existing usernames in the database, or if no user is in production yet, the existing migration should be updated.

**Comment by kernicPanel** [comment]
> IMHO, a migration should downcase existing usernames in the database, or if no user is in production yet, the existing migration should be updated.

Sorry, I didn't think about it before

**Comment by sampaccoud** [comment]
> IMHO, a migration should downcase existing usernames in the database, or if no user is in production yet, the existing migration should be updated.

Good point ! But this project is not released yet so we are still in the phase where we break the database in each PR!

**Review by kernicPanel — APPROVED** [review]
Don't forget to update the migrations with the new text helper.

**Pull request merged**